haystack/agents/conversational.py

```diff
@@ -1,63 +1,122 @@
+    tools = [Tool(name="ExampleTool", pipeline_or_node=example_tool_node)]
 
-    pn = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
-    agent = ConversationalAgent(pn, memory=ConversationSummaryMemory(pn))
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node, tools=tools)
 
+    # Use the agent in a chat application
     while True:
         user_input = input("Human (type 'exit' or 'quit' to quit): ")
         if user_input.lower() == "exit" or user_input.lower() == "quit":
             break
-        elif user_input.lower() == "memory":
-            print("\nMemory:\n", agent.memory.load())
         else:
             assistant_response = agent.run(user_input)
             print("\nAssistant:", assistant_response)
+    ```
 
+    If you don't want to have any tools in your chat app, you can create a ConversationalAgent only with a PromptNode:
+    ```python
+    import os
+
+    from haystack.agents.conversational import ConversationalAgent
+    from haystack.nodes import PromptNode
+
+    # Initialize a PromptNode
+    prompt_node = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node)
     ```
+
+    If you're looking for more customization, check out [Agent](https://docs.haystack.deepset.ai/reference/agent-api).
     """
 
-    def __init__(
-        self,
-        prompt_node: PromptNode,
-        prompt_template: Optional[Union[str, PromptTemplate]] = None,
-        memory: Optional[Memory] = None,
-        prompt_parameters_resolver: Optional[Callable] = None,
-    ):
+    def __init__(self, prompt_node: PromptNode, tools: Optional[List[Tool]] = None, memory: Optional[Memory] = None):
         """
-        Creates a new ConversationalAgent instance
-
-        :param prompt_node: A PromptNode used to communicate with LLM.
-        :param prompt_template: A string or PromptTemplate instance to use as the prompt template. If no prompt_template
-        is provided, the agent will use the "conversational-agent" template.
-        :param memory: A memory instance for storing conversation history and other relevant data, defaults to
-        ConversationMemory.
-        :param prompt_parameters_resolver: An optional callable for resolving prompt template parameters,
-        defaults to a callable that returns a dictionary with the query and the conversation history.
+        Creates a new ConversationalAgent instance.
+
+        :param prompt_node: A PromptNode used by Agent to decide which tool to use and what input to provide to it
+        in each iteration. If there are no tools added, the model specified with PromptNode will be used for chatting.
+        :param tools: A list of tools to use in the Agent. Each tool must have a unique name.
```

Comment:
If the ConversationalAgent has tools as a parameter, one could argue that also the standard Agent should accept tools instead of a tools_manager.

Reply:
`ToolsManager` accepts two parameters: `tools` and `tool_pattern`. For the ConversationalAgent, we don't need to change the default value of `tool_pattern` but for Agent, this might be required to be customized according to the prompt.

Reply:
Okay, I better understand now that you envision the ConversationalAgent as a special kind of Agent with predefined parameter settings. If a user wants to customize a ConversationalAgent, they would need to create an Agent and copy some of the settings from the ConversationalAgent and customize the rest of the settings.
For tools/tools_manager, I would still argue that the Agent and the ConversationalAgent should have the same parameter with the same name. So if we don't want to have `tools` as a parameter for both (which I see still as an option), then why not have `tools_manager` for both? Initialization of a ConversationalAgent could contain `tools_manager=ToolsManager([your_tool]` then and None as default.
We have that in our Agent example [here](https://github.com/deepset-ai/haystack/blob/main/examples/agent_multihop_qa.py#L100) and it would be great if users can replace Agent with ConversationalAgent with minimal changes.

Reply:
I agree with you about "having the same parameter with the same name" for Agent and ConversationalAgent. However, this would mean adding complexity to ConversationalAgent by exposing ToolsManager. My approach would be to have a component as simple as possible but happy to follow any convention we have

Reply:
It is required in the prompt, but it needs to be parsed by the ToolsManager. TM must know how to extract tool input and tool names from the toolbox. What you want to do is a logical and sound direction, precisely what we wanted to do from the beginning, but we need to specify this regex pattern somewhere, and the most logical place was ToolsManager. See this [discussion](https://github.com/deepset-ai/haystack/pull/4794#discussion_r1183514315) for more details.

Reply:
Ok, I understand what you are saying now. I just realized that ConversationalAgent couldn't change the template!